bluesky_kafka/tests/test_kafka.py

```diff
@@ -1,94 +1,202 @@
         def put_in_queue(name, doc):
-            print('putting ', name, 'in queue')
+            logger = logging.getLogger("bluesky.kafka")
+            logger.debug("putting %s in queue", name)
             queue.put((name, doc))
 
-        kafka_dispatcher = RemoteDispatcher('127.0.0.1:9092', group_id='kafka-unit-test')
+        kafka_dispatcher = RemoteDispatcher(
+            topics=[TEST_TOPIC],
+            bootstrap_servers=bootstrap_servers,
+            group_id="kafka-unit-test-group-id",
+            # "latest" should always work but
+            # has been failing on Linux, passing on OSX
+            consumer_config={"auto.offset.reset": auto_offset_reset},
+            polling_duration=1.0,
+            deserializer=deserializer,
+        )
         kafka_dispatcher.subscribe(put_in_queue)
         kafka_dispatcher.start()
 
-    queue = multiprocessing.Queue()
-    dispatcher_proc = multiprocessing.Process(target=make_and_start_dispatcher,
-                                              daemon=True, args=(queue,))
+    queue_ = multiprocessing.Queue()
+    dispatcher_proc = multiprocessing.Process(
+        target=make_and_start_dispatcher, daemon=True, args=(queue_,)
+    )
     dispatcher_proc.start()
     time.sleep(10)  # As above, give this plenty of time to start.
 
     local_accumulator = []
 
     def local_cb(name, doc):
-        print('local_cb: {}'.format(name))
+        print("local_cb: {}".format(name))
         local_accumulator.append((name, doc))
 
-    # Check that numpy stuff is sanitized by putting some in the start doc.
+    # Check that numpy data is sanitized by putting some in the start doc.
     # sending numpy arrays with md causes this:
     #    assert remote_accumulator == local_accumulator
     #    ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()
-
-    # md = {'stuff': {'nested': np.array([1, 2, 3])},
-    #     'scalar_stuff': np.float64(3),
-    #     'array_stuff': np.ones((3, 3))}
-    md = {}
+    # so the arrays are unpacked and compared explicitly
+    md = {
+        "numpy_data": {"nested": np.array([1, 2, 3])},
+        "numpy_scalar": np.float64(3),
+        "numpy_array": np.ones((3, 3)),
+    }
 
     RE.subscribe(local_cb)
     RE(count([hw.det]), md=md)
+    kafka_publisher.flush()
+    time.sleep(10)
 
     # Get the documents from the queue (or timeout --- test will fail)
     remote_accumulator = []
     for i in range(len(local_accumulator)):
-        remote_accumulator.append(queue.get(timeout=2))
+        remote_accumulator.append(queue_.get(timeout=2))
 
     dispatcher_proc.terminate()
     dispatcher_proc.join()
 
-    print('local_accumulator:')
+    print("local_accumulator:")
     pprint.pprint(local_accumulator)
-    print('remote_accumulator:')
+    print("remote_accumulator:")
     pprint.pprint(remote_accumulator)
 
-    # numpy arrays cause trouble sometimes
+    remote_np = remote_accumulator[0][1].pop("md")
```

Comment:
Instead of popping out `md` and `hints` for special treatment, I suggest using a pattern we have used elsewhere in this situation: apply [`event_model.sanitize_doc`](https://blueskyproject.io/event-model/api.html#event_model.sanitize_doc) to every document. (It serializes the whole thing to JSON and back, which takes care of both the numpy arrays and the list/tuple mismatch.) Unlike the current approach here, sanitizing is robust against future changes upstream; for example we might add new metadata to the plans that introduce tuples in places outside `hints`.

Reply:
Done!